### PR TITLE
Allow PHP8.3

### PR DESCRIPTION
**Pull request by heiglandreas**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

This allows usage of the package using PHP8.3

### CODE REVIEW

**Comment by Ocramius** [comment]
Lock file needs a hash refresh

**Comment by heiglandreas** [comment]
That's the least of all issues...

> 
> Reading package lists...
> Building dependency tree...
> Reading state information...
> E: Unable to locate package php8.3-apcu
> E: Couldn't find any package by glob 'php8.3-apcu'
> E: Unable to locate package php8.3-memcached
> E: Couldn't find any package by glob 'php8.3-memcached'
> E: Unable to locate package php8.3-mongodb
> E: Couldn't find any package by glob 'php8.3-mongodb'
> E: Unable to locate package php8.3-redis
> E: Couldn't find any package by glob 'php8.3-redis'

https://github.com/laminas/laminas-diagnostics/actions/runs/6954575541/job/18921670184?pr=74

**Comment by Ocramius** [comment]
Ah yes, I think we solved that recently for laminas-session?

**Comment by froschdesign** [comment]
See: https://github.com/laminas/laminas-session/blob/e733d986b30704831c6ec530d06688faf6f0856a/.laminas-ci/install-mongodb-extension-via-pecl.sh

**Comment by heiglandreas** [comment]
Stupid question perhaps: Why are we going through all the hassle of setting up Memcached and Redis and Mongo et all to then just ignore those tests? 

```
There were 10 skipped tests:

1) LaminasTest\Diagnostics\ChecksTest::testRabbitMQ
RabbitMQ tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:100

2) LaminasTest\Diagnostics\ChecksTest::testRedis
Redis tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:117

3) LaminasTest\Diagnostics\ChecksTest::testMemcache
Memcache tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:135

4) LaminasTest\Diagnostics\ChecksTest::testMemcached
Memcached tests are not enabled; enable them in phpunit.xml

/github/workspace/test/ChecksTest.php:152

5) LaminasTest\Diagnostics\ChecksTest::testMongo
Mongo tests are not enabled; enable them in phpunit.xml
```

**Comment by Xerkus** [comment]
That got left behind when moving to current common CI setup I would guess and noone paid enough attention to notice.

File: src/Check/IniFile.php [hunk 1]
```diff
@@ -22,7 +22,7 @@ class IniFile extends AbstractFileCheck
      */
     protected function validateFile($file)
     {
-        if (! is_array($ini = parse_ini_file($file)) || count($ini) < 1) {
+        if (! is_array($ini = @parse_ini_file($file)) || count($ini) < 1) {
```
Comment (Ocramius) [inline src/Check/IniFile.php]: Could we use something like `set_error_handler()` + `restore_error_handler()` here, perhaps? :thinking: 

Using `@` always bites us back later, so just wondering

**Comment by heiglandreas** [comment]
OK. Running the actual tests with the external tools fails. But it at least fails in all tested PHP-versions, so it's not related to PHP8.3

I created an issue to reintroduce these tests at #75

File: src/Check/IniFile.php (hunk 1 above)
Comment (heiglandreas) [reply src/Check/IniFile.php]: I was indeed thinking about that and decided to skip it here for sake of brevity. But using the error handler might actually allow us to add the reported message to the error-message.

So will add

**Comment by heiglandreas** [comment]
So shall I hit the "Merge" Button? Or is there anything missing?

**Comment by Chris53897** [comment]
Thanks. Is there a release date planned?

**Pull request merged**